public/app/user-service-registry/user-service-registry.controller.js

```diff
@@ -21,86 +21,437 @@
+        };
+
+        $scope.describeService = function() {
+
+            $scope.errorMsg = "";
+            var job = {
+                "apiKey": "my-api-key-38n987",
+                "jobType" : {
+                    "type": "read-service",
+                    "serviceID" : $scope.serviceId
+                }
+            };
+            var fd = new FormData();
+            fd.append( 'body', angular.toJson(job) );
+            var request = $http({
+                method: "POST",
+                url: '/proxy?url=pz-gateway.stage.geointservices.io/job',
+                data :fd,
+                headers: {"Content-Type": undefined}
+            }).then(function successCallback( html ) {
+                $scope.jobId = html.data.jobId;
+                $scope.DescribeServiceRetries = 0;
+                getDescribeServiceResult($scope.jobId)
+
+
+            }, function errorCallback(response){
+                console.log("user-service-registry.controller fail");
+                toaster.pop('error', "Error", "There was an issue with your request.");
             });
+
+        }
+        $scope.registerService = function() {
+            $scope.errorMsg = "";
+
+
+            var resourceMetadata = {
+                "name":$scope.serviceName,
+                "description":$scope.serviceDescription,
+                "url":$scope.serviceUrl,
+                "method":$scope.method
+            };
+
+            var data = {
+                "resourceMetadata" : resourceMetadata,
+                "inputs" : $scope.inputs,
+                "outputs" : $scope.outputs
+            };
+            var job = {
+                "apiKey": "my-api-key-38n987",
+                "jobType" : {
+                    "type": "register-service",
+                    "data" : data
+                }
+            };
+
+            var fd = new FormData();
+            fd.append( 'body', angular.toJson(job) );
+
+
+
+
+            var request = $http({
+                method: "POST",
+                url: '/proxy?url=pz-gateway.stage.geointservices.io/job',
+                data :fd,
+                headers: {"Content-Type": undefined}
+            }).then(function successCallback( html ) {
+               $scope.jobId = html.data.jobId;
+                $scope.RegisterResultsRetries = 0;
+                getRegisterResult($scope.jobId)
+
+
+            }, function errorCallback(response){
+                console.log("user-service-registry.controller fail");
+                toaster.pop('error', "Error", "There was an issue with your request.");
+            });
+
         };
 
+        function createExecuteInputMap(inputs) {
+            //executeInputMap
+            var i = 0;
+            for (i = 0; i < inputs.length; i++) {
+                switch (inputs[i].dataType.type) {
+                    // TODO: Looks like this can be simplified because body, raster and text all do the same thing
```

Comment:
This line can be removed now that it's fixed.  I think i meant to either comment in the PR or in the code and i ended up doing both.  Sorry